packages/components/src/DataList/DataList.tsx

```diff
@@ -39,6 +44,20 @@ export function DataList<T extends DataListObject>({
   return (
     <div className={styles.wrapper}>
       {/* List title and counter */}
+      <div className={styles.titleContainer}>
+        {title && <Heading level={3}>{title}</Heading>}
+        {totalCount !== null && (
```

Comment:
curious. what is this logic trying to do?

If it's undefined, it's gonna go through and render a (0 results) text or a glimmer.